From 9a0586b32808f404f0f1541d5fd69a984b8de360 Mon Sep 17 00:00:00 2001
From: yangerkun <yangerkun@huawei.com>
Date: Wed, 29 Mar 2023 20:11:43 +0800
Subject: [PATCH 360/921] [Huawei] Revert "[Backport] dm-crypt: fix softlockup
 in dmcrypt_write"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188393

-------------------------------

This reverts commit fcd26cca1b62b049d1382bb3baaac094ba1f7561.

Revert local patch, will fix this problem with mainline patch.

Signed-off-by: yangerkun <yangerkun@huawei.com>
---
 drivers/md/dm-crypt.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/drivers/md/dm-crypt.c b/drivers/md/dm-crypt.c
index b790365cb61b..3d975db86434 100644
--- a/drivers/md/dm-crypt.c
+++ b/drivers/md/dm-crypt.c
@@ -1892,7 +1892,6 @@ static int dmcrypt_write(void *data)
 {
 	struct crypt_config *cc = data;
 	struct dm_crypt_io *io;
-	unsigned long start_time = jiffies;
 
 	while (1) {
 		struct rb_root write_tree;
@@ -1915,7 +1914,6 @@ static int dmcrypt_write(void *data)
 
 		schedule();
 
-		start_time = jiffies;
 		set_current_state(TASK_RUNNING);
 		spin_lock_irq(&cc->write_thread_lock);
 		goto continue_locked;
@@ -1927,10 +1925,6 @@ static int dmcrypt_write(void *data)
 
 		BUG_ON(rb_parent(write_tree.rb_node));
 
-		if (time_is_before_jiffies(start_time + HZ)) {
-			cond_resched();
-			start_time = jiffies;
-		}
 		/*
 		 * Note: we cannot walk the tree here with rb_next because
 		 * the structures may be freed when kcryptd_io_write is called.
-- 
2.17.1

